src/metrics.jl

```diff
@@ -73,20 +73,11 @@ function binary_statistics(confusion::AbstractMatrix, class_index::Integer)
     false_positives = predicted_positives - true_positives
     false_negatives = actual_positives - true_positives
     true_negatives = actual_negatives - false_positives
-    true_positive_rate = (true_positives == 0 && actual_positives == 0) ?
-                         (one(true_positives) / one(actual_positives)) :
-                         (true_positives / actual_positives)
-    true_negative_rate = (true_negatives == 0 && actual_negatives == 0) ?
-                         (one(true_negatives) / one(actual_negatives)) :
-                         (true_negatives / actual_negatives)
-    false_positive_rate = (false_positives == 0 && actual_negatives == 0) ?
-                          (zero(false_positives) / one(actual_negatives)) :
-                          (false_positives / actual_negatives)
-    false_negative_rate = (false_negatives == 0 && actual_positives == 0) ?
-                          (zero(false_negatives) / one(actual_positives)) :
-                          (false_negatives / actual_positives)
```

Comment:
Do you remember there a rationale behind these quantities being 1/0 in the edge case? I agree these should return NaN but just curious

Reply:
I traced this code back to https://github.com/beacon-biosignals/OldLighthouse.jl/pull/36 (private repo), so I believe it was to support ROC curves better, where NaNs could contaminate the whole AUC even if it's just some 0.0 threshold or something. But IMO that is much better handled in the AUC computation itself rather than here.